lib/index.ts

```diff
@@ -12,18 +12,18 @@ export enum ProcessDataFlag {
   CommandLine = 2
 }
 
-interface  IRequest {
-  callback: (processes: IProcessTreeNode | IProcessInfo[] | undefined) => void;
+interface  IRequest<T> {
+  callback: (result: T) => void;
```

Comment:
It's probably better to make this `T | undefined` and remove it below

Reply:
I disagree — that would weaken this interface.

The ` … | undefined` type requirement isn't imposed by the callback, it's imposed by the need for the callback to handle the return type of the individual transforms; and both of them happen to include `undefined` in their codomain.

For example, it prevents describing a transformation which returns some other error type instead (e.g. a `Promise` which might be rejected). It also infects both `getRawProcessList`'s parameters (the callback and the transform result) with the same assumptions. It just gets messy faster, and does not add any type safety.